dashboard/lib/pd/survey_pipeline/daily_survey_decorator.rb

```diff
@@ -136,6 +139,40 @@ def self.data_visible_to_user?(user, data)
       true
     end
 
+    # Given metadata of a survey submission, returns context of that submission.
+    # Example return values: Pre Workshop, Post Workshop, Day <number>, and Facilitator.
+    #
+    # @param workshop_id [Number] must be valid workshop id
+    # @param day [Number]
+    # @param facilitator_id [Number]
+    # @param form_id [Number]
+    #
+    # @return [String] context name
+    #
+    def self.get_survey_context(workshop_id, day, facilitator_id, form_id)
+      workshop = Pd::Workshop.find_by_id(workshop_id)
+      return 'Invalid' unless workshop && day >= 0
+
+      if workshop.csf?
+        return 'Facilitators' if facilitator_id
+
+        # CSF is a 1-day workshop and doesn't have daily survey.
+        return day == 0 ? 'Pre Workshop' : 'Post Workshop'
+      elsif workshop.summer?
+        # Summer workshop doesn't have post-workshop survey.
+        return day == 0 ? 'Pre Workshop' : "Day #{day}"
+      else
+        # Academic year workshop doesn't have pre-workshop survey.
+        return 'Invalid' if day == 0
+
+        # Post workshop survey has the same "day" value as the last daily survey.
+        # Use form name to distinguish them.
+        session_count = workshop.sessions.size
+        form_name = get_form_short_name(form_id)
+        return day == session_count && form_name == 'Post Workshop' ? 'Post Workshop' : "Day #{day}"
```

Comment:
This feels like code we'll need to modify to support future survey types.  Any thoughts now on ways we can drive this sort of tab-naming with configuration in the future instead of hard-coding it?

Reply:
Yes, one solution is to add `context_name` or `tag` field to every survey submissions. For JotForm survey embedded in our site, we should know exactly what context that survey if for and send it along to JotForm. 

To go one step further, we can combine individual fields like `workshop_id`, `facilitator_id`, `form_id` and `context_name` into 1 hash object, which is then passed from our site to JotForm then saved to database later when we sync JotForm submissions. By doing this we can use 1 model to save all survey submissions.